### PR TITLE
Don't wrap RayError with RayTaskError

### LINKED ISSUE #5862: RayTaskError doesn't subclass original exception correctly
### System information
- **Ray version**: 0.7.5

### Describe the problem
I'm not exactly sure what's going on, but it looks like some issue with recreating the original exception. The original exception here is a `ray.exceptions.UnreconstructableError`, but it fails when trying to print the exception because there is no `self.object_id`.

### Source code / logs
```python
import numpy as np
import ray
ray.init(object_store_memory=1e9)

@ray.remote
def f(x):
    return x

# Trigger unreconstructable object error.
x = [f.remote(np.ones(int(1e7))) for _ in range(10)]
ray.get(x[0])
```

This results in:
```
Traceback (most recent call last):
  File "/Users/swang/ray/python/ray/worker.py", line 2121, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(UnreconstructableError): <exception str() failed>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "error.py", line 11, in <module>
    ray.get(x[0])
  File "/Users/swang/ray/python/ray/worker.py", line 2131, in get
    return values
  File "python/ray/includes/libcoreworker.pxi", line 32, in ray._raylet.ProfileEvent.__exit__
    "value": str(value),
  File "/Users/swang/ray/python/ray/exceptions.py", line 155, in __str__
    self.object_id.hex(),
AttributeError: 'RayTaskError(UnreconstructableError)' object has no attribute 'object_id'
2019-10-08 10:35:18,456 WARNING worker.py:1619 -- The task with ID ffffffffffffffffffff01000000 is a driver task and so the object created by ray.put could not be reconstructed.
```

It looks like it's related to # #5799.

### PR DESCRIPTION
**Pull request by ericl**
Closes https://github.com/ray-project/ray/issues/5862

### CODE REVIEW

**Comment by ericl** [comment]
ok to test

File: python/ray/exceptions.py [hunk 1]
```diff
@@ -56,6 +56,9 @@ def as_instanceof_cause(self):
         if issubclass(RayTaskError, self.cause_cls):
             return self  # already satisfied
 
+        if issubclass(RayError, self.cause_cls):
```
Comment (stephanie-wang) [inline python/ray/exceptions.py]: Hmm this doesn't seem to fix #5862 for me, but it does work if I do `issubclass(self.cause_cls, RayError)`.

Comment (ericl) [reply python/ray/exceptions.py]: How about now?

**Review by pcmoritz — APPROVED** [review]

File: python/ray/exceptions.py (hunk 1 above)
Comment (stephanie-wang) [reply python/ray/exceptions.py]: Looks good. Thanks!

**Pull request merged**